relay-server/src/services/project.rs

```diff
@@ -702,8 +721,18 @@ impl Project {
         match &mut self.state {
             State::Cached(state) => {
                 let state = Arc::clone(state);
+                let mode = state.get_extraction_mode();
+
+                let buckets = self.filter_buckets(
+                    outcome_aggregator.clone(),
+                    metric_stats.clone(),
+                    mode,
+                    &state,
+                    buckets,
+                );
 
-                let buckets = self.rate_limit_metrics(&state, buckets, outcome_aggregator.clone());
+                let buckets =
+                    self.rate_limit_buckets(outcome_aggregator.clone(), mode, &state, buckets);
```

Comment:
Please double check me here:

It seems we only call `rate_limit_buckets` when we also call `rate_limit_and_merge_buckets` right after, latter also checks the cached limits but for processing Relays also forwards to the consistent rate limiter -> we can remove `rate_limit_buckets` as it is not needed.

This also allows us to only `filter_buckets` when merging into the aggregator in `rate_limit_and_merge_buckets` which we should rename to something better.